modules/oe_subscriptions_anonymous/tests/src/Kernel/AccessCheckTest.php

```diff
@@ -0,0 +1,185 @@
+    // Services a variables.
+    $route_name = 'oe_subscriptions_anonymous.anonymous_subscribe';
+    // Create a flag.
+    $flag_id = 'subscribe_article';
+    $flag = $this->createFlagFromArray([
+      'id' => $flag_id,
+      'flag_type' => $this->getFlagType('node'),
+      'entity_type' => 'node',
+      'bundles' => ['article'],
+    ]);
+    // A flag that applies to all bundles.
+    $this->createFlagFromArray([
+      'id' => 'another_flag',
+      'flag_type' => $this->getFlagType('entity_test_with_bundle'),
+      'entity_type' => 'entity_test_with_bundle',
+      'bundles' => [],
+    ]);
+    // Node.
+    $node = $this->createNode([
+      'type' => 'article',
+      'status' => 1,
+    ]);
+    $node->save();
+    $nid = $node->id();
+    $access_manager = $this->container->get('access_manager');
+    // Give access content permission to anonymous.
+    $this->grantPermissions(Role::load('anonymous'), ['access content']);
+    $this->setCurrentUser(new AnonymousUserSession());
+    // Empty.
+    $access = $access_manager->checkNamedRoute(
+      $route_name,
+      [
+        'flag' => '',
+        'entity_id' => '',
+      ],
+      NULL,
+      TRUE,
+    );
+    $this->assertFalse($access->isAllowed());
+
+    // No matching flag.
+    $access = $access_manager->checkNamedRoute(
+      $route_name,
+      [
+        'flag' => 'subscribe_events',
+        'entity_id' => $nid,
+      ],
+      NULL,
+      TRUE,
+    );
+    $this->assertFalse($access->isAllowed());
+
+    // Flag doesn't starts with 'subscribe_'.
+    $access = $access_manager->checkNamedRoute(
+      $route_name,
+      [
+        'flag' => 'another_flag',
+        'entity_id' => $nid,
+      ],
+      NULL,
+      TRUE,
+    );
+    $this->assertFalse($access->isAllowed());
+
+    // Disabled flag.
+    $flag->disable();
+    $flag->save();
+    $access = $access_manager->checkNamedRoute(
+      $route_name,
+      [
+        'flag' => $flag_id,
+        'entity_id' => $nid,
+      ],
+      NULL,
+      TRUE,
+    );
+    $this->assertFalse($access->isAllowed());
+    $flag->enable();
+    $flag->save();
+
+    // Not existing node.
+    $access = $access_manager->checkNamedRoute(
+      $route_name,
+      [
+        'flag' => $flag_id,
+        'entity_id' => '1234',
+      ],
+      NULL,
+      TRUE,
+    );
+    $this->assertFalse($access->isAllowed());
+
+    // Finally a subscribe parameters matching all conditions.
+    $access = $access_manager->checkNamedRoute(
+      $route_name,
+      [
+        'flag' => $flag_id,
+        'entity_id' => $nid,
+      ],
+      NULL,
+      TRUE,
+    );
+    $this->assertTrue($access->isAllowed());
+
+    // Route is not allowed for logged users.
+    $access = $access_manager->checkNamedRoute(
+      $route_name,
+      [
+        'flag' => $flag_id,
+        'entity_id' => $nid,
+      ],
+      $this->createUser([], 'logged_user'),
+      TRUE,
+    );
+    $this->assertFalse($access->isAllowed());
+
+  }
```

Comment:
Let's add an helper method to this test to improve readability.
`protected function assertRouteAccess(string $flag_id, string $entity_id, AccountInterface $user = NULL)`. So you just pass the minimum needed information and it becomes more readable.

Reply:
Done